OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -226,6 +227,9 @@ protected override void OnFirstRun(Actor self)
 			if (target.IsValidFor(self))
 			{
 				QueueChild(new Fly(self, target, target.CenterPosition));
+
+				// Fly a single tick forward so we have passed the target and start flying out of range facing away from it
```

Comment:
This won't necessarily be true, depending on the relative direction and speed of the two units. What will happen if this assumption fails?

Reply:
IIrc: It'll fly "backwards" away from the target (still facing it if the aircraft can slide) which is the trade-off of using @tovl's suggestion here instead of always flying the distance straight. The other option I can think of would be giving `FlyStep` a target which it tries to face while flying the step.

Reply:
If the assumption fails, it won't even reach this stage because it is still trying to catch up with the target?

Tbh, I don't think we should be breaking our heads over edge cases like this now, when #18386 and its predecessors will be changing the underlying assumptions anyway.

Reply:
>  it won't even reach this stage because it is still trying to catch up with the target?

Again, this assumes that both units are moving in the same direction and do not turn. I raised the point specifically because "target changes direction while moving" is the kind of thing that is easily forgotten in constructed test cases but happens all the time during real gameplay.

Reply:
This specific addition can't make things any worse than the current behaviour, so its not worth blocking this over it.